### DISCUSSION TITLE
3.9.0beta1 New Client Level behaviour

### DISCUSSION THREAD

**Discussion by rdica** [jamulussoftware/jamulus#2725] [Bug Investigation]
I started up a fresh instance of 3.9.0beta1 (compiled from src on Ubuntu 18.04) with no existing ini file, set a username, stereo out, new client level at 0%, connected to a server with players on it (Jamsuckers 212.83.56.10 on Any Genre 1, running version 3.8.2dev-1646e905 on Linux).

I expected no audio as new client was at 0% and indeed all the sliders were at 0 yet there was audio. I double checked to make sure I hadn't accidently looped a jack port mapping but there wasn't one. I further tested this behaviour against other servers with players on them, deleting the ini file and starting from scratch with the same settings each time. Server versions tested against were 3.8.1 and 3.8.2dev-c5111db6-dirty. All had the same client behaviour, there was audio even though all players sliders were at 0%.
I repeated above procedures using 1% and got the reduced volumes expected, which were quite lower in volume than what I heard when new client level was at 0%.

I tested same procedure with 3.8.2 (compiled from src on Ubuntu 18.04) against the same servers and got the expected behaviour of no audio when new client level was set to 0%.

Bug?

**Comment by pljones** [comment]
Did you try:
* mute checkbox
* dragging the slider to the bottom

and comparing with the new client level 0%?  It'll help pin down where the problem is.

**Reply by rdica** [reply]
Joined a server, ensured every active players slider was at the bottom (they are, and I could still hear them), muted each player, unmuted each player, could hear only the last unmuted player. Muted/unmuted that one again, all was now quiet. Individually increased each active players slider and could begin to hear them as expected.

New players that joined after this could not be heard, seems the new client level was correctly applied.

**Reply by rdica** [reply]
I forgot to mention for completeness, I have the RPC functionality enabled though no operations were performed using the api.

Running 
`/home/ardy/src/jamulus-r3_9_0beta1/jamulus -6 -j --clientname RPC --jsonrpcport 55124 --jsonrpcsecretfile /home/ardy/rpcfile-jamulus -i /home/ardy/.config/Jamulus/Jamulus-RPC.ini`

**Reply by pljones** [reply]
> muted each player, unmuted each player, could hear only the last unmuted player.

That sounds like how solo would work.  Are you sure you used mute?  Or am I misunderstanding you?

**Reply by rdica** [reply]
Yes, I'm sure. I'm simply reporting what I heard after the last unmute.

I just tested again, fresh ini, join server, mute/unmute active players, this time the last unmute wasn't audible and their volume sliders worked as expected.
A player that wasn't active during the initial mute/unmutes, but was on the server when I first joined, started playing. Though their slider was at zero they were still audible. A mute/umute seemed to reset his channel as well.

**Comment by rdica** [comment]
Here's a screenshot of jamulus client, qjackctl patchbay, and jack_mixer indicating audio levels. There should be no level showing in jack_mixer as all sliders in jamulus are at 0.

![jamulus-390beta1-jack](https://user-images.githubusercontent.com/16609801/178158070-3d65d331-7f0c-47f7-bbbb-413d9a7722f2.png)

**Reply by pljones** [reply]
It's hard to read on that screenshot.  What's connected to what?

**Reply by rdica** [reply]
Apologies, for the purpose of showing audio signal path, `Jamulus RPC:output left` and `Jamulus RPC:output right` connect to `jack_mixer:deck1 L` and `jack_mixer:deck1 R` respectively.
Looking at the jack_mixer UI at bottom left, you'll see a signal on the `deck1` channel, indicating audio coming from the jamulus client.
Every user in the Jamulus client is at zero so there shouldn't be any signal showing in jack_mixer `deck1` channel.

**Comment by rdica** [comment]
Further testing using 1% as new client level on a fresh execution (no exisiting ini) shows same behaviour as initial report.

**Comment by hoffie** [comment]
Thanks for the report. I can confirm this behavior and can also confirm that 3.8.2 did not show it, but `master` does. I've opened #2730 based on this discussion and added further ideas to investigate. I'm not actively working on it yet though, but it should be fixed before 3.9.0 final.

**Reply by hoffie** [reply]
With #2737 merged, this should now be fixed on `master`. Thanks again for catching this during the beta phase! :)
